Fixes #151: Add tomof to cimproperty and connect to ciminstance

KSchopmeyer opened the pull request:
Adds function tomof() to CIMProperty instead of burying code with CIMInstance.  Will connect this to CIMClass in future (issue filed).

Ready for review.

andy-maier commented:
-1

Comments:

* I don't agree with the comment "# this should never happen" on line 2499: `None` is the normal representation of NULL.
* The docstring of `tomof()` should be adjusted to the Google format. See e.g. `tomof()` in `CIMParameter`.
* Update the change log (`docs/changes.rst`).

KSchopmeyer commented:
1. You are right.  I reorged the code several times and at one point I tested null before the call.  Removed.

2. Docstring.  Corrected.

3. Done.

Note that the isInstance variable is not used yet.  Will finish that in separate pr for issue #223 or new issue.
